src/app/shared/components/xterm-terminal/component.ts

```diff
@@ -0,0 +1,252 @@
+  }
+
+  initSubscriptions(): void {
+    this.projectId = this._activatedRoute.snapshot.paramMap.get(PathParam.ProjectID);
+    this.clusterId = this._activatedRoute.snapshot.paramMap.get(PathParam.ClusterID);
+
+    // Note:
+    // Get Cluster Info inorder to show cluster name in shell
+    if (this.projectId && this.clusterId) {
+      this._clusterService
+        .cluster(this.projectId, this.clusterId)
+        .pipe(take(1))
+        .subscribe((cluster: Cluster) => {
+          this.cluster = cluster;
+          this.initTerminal();
+        });
+    }
+
+    // Subscription bidirectional communication
+    this._websocketService.messages$.pipe(takeUntil(this._unsubscribe)).subscribe((data: ITerminalFrame) => {
+      this._handleWebSocketConnectionMessages(data);
+    });
+
+    // Subscription connection success
+    this._websocketService
+      .getWebSocketOnConnectObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this._notificationService.success('Successfully connected to cloud shell terminal');
+        this.isConnectionLost = false;
+      });
+
+    // Subscription connection lost
+    this._websocketService
+      .getWebSocketOnCloseObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        if (this.terminal) {
+          this.terminal.setOption('cursorBlink', false);
+        }
+        this.isConnectionLost = true;
+      });
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+    this._websocketService.close();
+
+    // Clean 'resize' Event listener subscription
+    this._removeEventListener();
+  }
+
+  ngAfterViewInit(): void {
+    this._connectToWebSocketConnection();
+  }
+
+  initTerminal(): void {
+    if (this.terminal) {
+      this.terminal.dispose();
+    }
+
+    this.terminal = new Terminal({
+      fontSize: Config.fontSize,
+      fontFamily: Config.fontFamily,
+      cursorBlink: Config.cursorBlink,
+      theme: {
+        foreground: Config.foreground,
+        background: Config.background,
+      },
+    });
+
+    const containerElement = this.terminalRef.nativeElement;
+    const fitAddon = new FitAddon();
+    this.terminal.loadAddon(fitAddon);
+    this.terminal.open(containerElement);
+
+    const clusterName = this.cluster && this.cluster.name;
+    this.terminal.write('\x1b[37mWelcome to Cloud Shell! Type "help" to get started.\r\n');
+    this.terminal.write(
+      `\x1b[37mYour Cloud Platform project in this session is set to \x1b[1;34m${clusterName}\x1B[0m\n\n\r`
+    );
+
+    const delayFn = debounce(() => {
+      fitAddon.fit();
+    }, this.DELAY_TIMEOUT);
+    delayFn();
+
+    this._removeEventListener();
+    // Event listeners binding
+    this._resizeEventlistenerFn = this._renderer.listen(window, 'resize', () => {
+      delayFn();
+    });
+
+    // XTerminal Properties
+    this.terminal.onData(this._onTerminalSendString.bind(this));
```

Comment:
can we replace it to `this.terminal.onData(() => this._onTerminalSendString.bind());` ?

Reply:
2 points here:
- `bind` takes `this` argument which is missing.
- we can use like this passing as a `arrow` function call back which is `lexically` using parent `this` no need to use `this` bind so I just used shorter syntax. 

```ts
    this.terminal.onData(event => {
      this._onTerminalSendString(event);
    });
```